src/main/java/mate/academy/hw05/map/ReferenceBook.java

```diff
@@ -0,0 +1,237 @@
+        hashTable = new Node[oldHashTable.length * 2];
+        size = 0;
+        for (Node<K, V> node : oldHashTable) {
+            if (node != null) {
+                for (Node<K, V> n : node.getNodes()) {
+                    insert(n.key, n.value);
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean delete(final K key) {
+        int index = hash(key);
+        if (hashTable[index] == null)
+            return false;
+
+        if (hashTable[index].getNodes().size() == 1) {
+            hashTable[index].getNodes().remove(0);
+            return true;
+        }
+
+        List<Node<K, V>> nodeList = hashTable[index].getNodes();
+        for (Node<K, V> node : nodeList) {
+            if (key.equals(node.getKey())) {
+                nodeList.remove(node);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public V get(final K key) {
+        int index = hash(key);
+        if (index < hashTable.length &&
+                hashTable[index] != null) {
+            
+            if (hashTable[index].getNodes().size() == 1) {
+                return hashTable[index].getNodes().get(0).getValue();
+            }
+
+            List<Node<K, V>> list = hashTable[index].getNodes();
+            for (Node<K, V> node : list) {
+                if (key.equals(node.getKey())) {
+                    return node.getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    private int hash(final K key) {
+        int hash = 31;
+        hash = hash * 17 + key.hashCode();
+        return hash % hashTable.length;
+    }
+
+    @Override
+    public Iterator<V> iterator() {
+        return new Iterator<V>() {
+            int counterArray = 0;
+            int valuesCounter = 0;
+            Iterator<Node<K, V>> subIterator = null;
+
+            @Override
+            public boolean hasNext() {
+                if (valuesCounter == size)
+                    return false;
+
+                if (subIterator == null || !subIterator.hasNext()) {
+                    if (moveToNextCell()) {
+                        subIterator = hashTable[counterArray].getNodes().iterator();
+                    } else {
+                        return false;
+                    }
+                }
+                return subIterator.hasNext();
+            }
+
+            private boolean moveToNextCell() {
+                counterArray++;
+                while (hashTable[counterArray] == null) {
+                    counterArray++;
+                }
+                return hashTable[counterArray] != null;
+            }
+
+            @Override
+            public V next() {
+                valuesCounter++;
+                return subIterator.next().getValue();
+            }
+        };
+    }
+
+    private class Node<K, V> {
+        private List<Node<K, V>> nodes;
```

Comment:
Node should not aggregate nodes, it can only contain reference to next. This is wrong because this models another data structure, not linked list.

Reply:
change everything related to getNodes()

Reply:
I did change